### ISSUE TITLE
Setting localStorage does not reflect in $localStorage

### ISSUE THREAD

**Issue by cyounkins** [gsklee/ngStorage#34]
It appears that setting localStorage in plain javascript is not picked up by ngStorage.

That is, when doing either

localStorage.setItem('key', 'value')  OR
localStorage['key'] = 'value'

changes are not propagated to the DOM.

Why would I want to do this? I'd like to persist part of localStorage across user logins by backing it up to the server and pushing it down to the client when necessary.

I think I could potentially wrap this in angular, but I'd prefer to not do that.

The solution may be easy too. Reading the code, it looks like there is an internal copy of the webStorage object that is saved in JS memory to compare it with at $watch intervals. Why not simply check against the webStorage object?

**Comment by egilkh** [comment]
Closing.

If any one still experience this problem please re-open.

**Comment by danielmeng07** [comment]
+1 Have this been fixed? This is extremely irritating that ngStorage does not update with native javascript localStorage. It seems backwards to me that if I want to change a simple login boolean in another tab, I have to import the entire angular library?

**Comment by arrowd** [comment]
I've got this problem in my app and, from what i've learned, it isn't ng-storage fault at all. When you do

```javascript
localStorage["foo"] = "bar";
// redirect to another page
```

that assignment to `localStorage` is executed asynchronously. That is, a browser really doesn't wait for the data to be written and proceeds to the new page. This interrupts the process of localStorage writing and hence you get empty storage. Here are some SO questions regarding this:

http://stackoverflow.com/q/21139572/637669
http://stackoverflow.com/a/35937656/637669

In short, there is nothing we can do except wrapping redirecting code into `setTimeout`.
